### PR TITLE
xmlsort: init at 1.01

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

I needed this.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Cc @volth

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2466,6 +2466,8 @@ in
     libmaxminddb = null;
   };
 
+  xmlsort = perlPackages.XMLFilterSort;
```
Comment (volth) [inline pkgs/top-level/all-packages.nix:2469]: maybe to `aliases.nix` ?

Comment (peterhoeg) [reply pkgs/top-level/all-packages.nix:2469]: I thought aliases was meant for when stuff was deprecated - not for introducing a package into the top level.

**Review by volth — APPROVED** [review]

**Comment by peterhoeg** [comment]
This is slightly off-topic @volth, but are you including all packages defined in perl-packages.nix into your automatic update jobs?

**Comment by stigtsp** [comment]
@peterhoeg Fails on `darwin x64`, `shortenPerlShebang $out/bin/xmlsort` is likely needed.

```
$ bin/xmlsort 
zsh: /nix/store/gh8lgp7kc6fbr6hk57m58lkin92sjsmp-perl5.32.0-XML-Filter-Sort-1.01/bin/xmlsort: bad interpreter: /nix/store/rqpmnkzxh5kgiykg9ff9v9n467g3i2r1-perl-5.32.0/bin/perl -w -I/nix/store/rqpmnkzxh5kgiykg9ff9v9n467g3i2r1-perl-5.32.0/: exec format error

```

**Comment by peterhoeg** [comment]
Nicely spotted, fixed.

@volth, should we maybe have the standard perl builder perform the shortening step on Darwin automatically for everything in $out/bin ?

**Comment by stigtsp** [comment]
> Nicely spotted, fixed. 

The commit to master e6b6f2109ff36fc3f7d6c0f046d3c14915c41d6c does not appear to have fixed this issue on darwin. Can you open a PR for this instead?

```
$ nix-build -A perlPackages.XMLFilterSort

Appending installation info to /nix/store/jfqr7q5innmd71dw8gsrknb3p9hiqqrp-perl5.32.0-XML-Filter-Sort-1.01/lib/perl5/5.32.0/darwin-thread-multi-2level/perllocal.pod
/nix/store/8h5awvnjzp0rgkby7s8fgzqg011mpzh7-stdenv-darwin/setup: line 89: shortenPerlShebang: command not found
builder for '/nix/store/4a3x3spyd4nissq7s8g1mr6flbmk8yrs-perl5.32.0-XML-Filter-Sort-1.01.drv' failed with exit code 127
error: build of '/nix/store/4a3x3spyd4nissq7s8g1mr6flbmk8yrs-perl5.32.0-XML-Filter-Sort-1.01.drv' failed
```

**Comment by stigtsp** [comment]
Some minor nitpicking:

- The commit message `xmlsort: init at 1.01` should probably have been `perlPackages.XMLFilterSort: init at 1.01`. I find it useful to be able to to parse the commit messages sometimes.

- There is no `meta` provided for this derivation.

**Comment by volth** [comment]
> This is slightly off-topic @volth, but are you including all packages defined in perl-packages.nix into your automatic update jobs?

Almost all.
There are few exceptions:
 * `SysVirt`, `MHonArc` and `AppSt` whose `src` does not point to CPAN
 * `MusicBrainz-DiscID-0.03`, `HTML-TableExtract-2.13`, `podlators-4.10`, `VM-EC2-1.28`, `libintl-perl-1.31`, `Inline-0.83`,  `libapreq2-2.13`, `Cpanel-JSON-XS-4.17`, `Perl-Tidy-20200110`, `DBIx-Class-0.082841` are pinned because newer versions fail to build
 * `HTTP-Daemon-6.01` are pinned because newer version fails to cross-compile
 * `Crypt-SSLeay-0.73_06` is newer than in CPAN index

**Comment by volth** [comment]
> Nicely spotted, fixed. @volth, should we maybe have the standard perl builder perform the shortening step on Darwin automatically for everything in $out/bin ?

OK.
The only obstacle is I do not have Darwin, only NixOS.
Maybe @stigtsp?

**Comment by stigtsp** [comment]
> [..] shortening step on Darwin automatically for everything in $out/bin ?

That's a good idea, was thinking to also check if including `perlPreHook = stdenv.lib.optionalString stdenv.isDarwin "export LD=$CC";` would also work.

**Pull request merged**